pymongo/client_session.py

```diff
@@ -947,9 +947,16 @@ def _txn_read_preference(self):
             return self._transaction.opts.read_preference
         return None
 
+    def _start_serv_sesh(self):
+        while isinstance(self._server_session, Empty):
+            self._server_session, old = self._client._get_server_session(), self._server_session
+            self._server_session.generation = old.generation
+            self._server_session.last_use = old.last_use
+            self._server_session._transaction_id = old._transaction_id
```

Comment:
You cannot overwrite a session's _transaction_id otherwise the session will have undefined behavior. You'll have to figure our another way to handle the `session._start_retryable_write()` behavior with an implicit session.